src/net_processing.cpp

```diff
@@ -1604,6 +1604,34 @@ void static ProcessGetBlockData(CNode* pfrom, const CChainParams& chainparams, c
     }
 }
 
+//! Determine whether or not a peer can request a transaction, and return a reference to it if so.
+Optional<CTransactionRef> static FindTxForGetData(CNode* peer, const CInv& inv, std::chrono::microseconds mempool_req, std::chrono::microseconds longlived_mempool_time) EXCLUSIVE_LOCKS_REQUIRED(cs_main)
```

Comment:
nit: could update `CInv& inv` to be `uint256& tx_hsh` since function just uses `inv` to look up the hash 3 times?

Reply:
also, is there a reason `mempool_req` & `longlived_mempool_time` aren't `const`?

Reply:
I agree with Amiti here.

Reply:
Done.

Reply:
Agree, done.